SCClassLibrary/Common/Math/SimpleNumber.sc

```diff
@@ -523,14 +523,21 @@ SimpleNumber : Number {
 	}
 
 	schedBundleArrayOnClock { |clock, bundleArray, lag = 0, server, latency|
-		clock.sched(this, {
-					if (lag != 0) {
-						SystemClock.sched(lag, {
-							server.sendBundle(latency ? server.latency, *bundleArray)
-						})
-					} {
-						server.sendBundle(latency ? server.latency, *bundleArray)
-					}
-		})
+		if (lag != 0) {
+			clock.sched(this, {
+				SystemClock.sched(lag, {
+					server.sendBundle(latency ? server.latency, *bundleArray)
+				})
+			})
+		} {
+			if(this != 0) {
+				clock.sched(this, {
+					server.sendBundle(latency ? server.latency, *bundleArray)
+				})
+			} {
+				server.sendBundle(latency ? server.latency, *bundleArray)
+			}
+		}
+
```

Comment:
if you rewrote this with the scheduling bit as a lambda, I think it would be a lot easier to read:

```supercollider
var schedFunc = { server.sendBundle(latency ? server.latency, *bundleArray) };
if(lag != 0) {
  clock.sched(this, { SystemClock.sched(lag, schedFunc) })
} {
  if(this != 0) {
    clock.sched(this, schedFunc)
  } {
    schedFunc.value
  }
}
```

Reply:
A comment explaining the different cases couldn't hurt, either